src/globus_sdk/transport/retry.py

```diff
@@ -0,0 +1,254 @@
+    return typing.cast(float, (0.25 + 0.5 * random.random()) * (2 ** ctx.attempt))
+
+
+# type var useful for declaring RetryPolicy
+RetryCheck = typing.Callable[[RetryContext], typing.Optional[bool]]
+
+
+class RetryPolicy:
+    """
+    The RetryPolicy object defines how retries are evaluated after a request.
+    It defines several default hooks which are executed after every request, and
+    additional hooks can be registered. The job of the RetryPolicy is to pass hooks
+    a RetryContext to evaluate. Once a hook has determined that the request should or
+    should not be retried, or all hooks have been consulted, the policy returns a
+    result.
+
+    The RetryPolicy is also responsible for determining the amount of backoff between
+    retries, and executing the sleeps between retries itself.
+
+    :param backoff: A function which determines how long to sleep between calls based on
+        the RetryContext. Defaults to expontential backoff with jitter based on the
+        context ``attempt`` number.
+    :type backoff: callable
+    :param checks: A list of initial checks for the policy. Any hooks registered,
+        including the default hooks, will run after these checks.
+    :type checks: list of callables
+    :param max_sleep: The maximum sleep time between retries (in seconds). If the
+        computed sleep time or the backoff requested by a retry check exceeds this
+        value, this amount of time will be used instead
+    :type max_sleep: int
+    :param max_retries: The maximum number of retries allowed by this policy. This is
+        checked by ``default_check_max_retries_exceeded`` to see if a request should
+        stop retrying.
+    :type max_retries: int
+    """
+
+    #: status codes for responses which may have a Retry-After header
+    RETRY_AFTER_STATUS_CODES = (429, 503)
+    #: status codes for error responses which should generally be retried
+    TRANSIENT_ERROR_STATUS_CODES = (429, 500, 502, 503, 504)
+    #: status codes indicating that authorization info was missing or expired
+    EXPIRED_AUTHORIZATION_STATUS_CODES = (401,)
+
+    def __init__(
+        self,
+        *,
+        backoff: typing.Callable[[RetryContext], float] = _exponential_backoff,
+        checks: typing.Optional[typing.List[RetryCheck]] = None,
+        max_sleep: int = 10,
+        max_retries: int = 5,
+    ):
+        self.backoff = backoff
+        self.max_sleep = max_sleep
+        self.max_retries = max_retries
+        # copy
+        self.checks = list(checks if checks else [])
+        # register internal checks
+        self.register_default_checks()
+
+    def compute_delay(self, ctx: RetryContext) -> float:
+        """
+        Given a retry context, compute the amount of time to sleep.
+        This is always the minimum of the backoff (run on the context) and the
+        ``max_sleep``.
+        """
+        delay = self.backoff(ctx)
+        return min(delay, self.max_sleep)
```

Comment:
I don't know what your preference is for this style, so I'll just point it out:
```suggestion
        return min(self.backoff(ctx), self.max_sleep)
```

Reply:
Sure. I have no opinion one way or another, so the fact that you're suggesting this is enough.